src/App.tsx

```diff
@@ -15,71 +16,131 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+  isSelected: boolean;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+function switcher(
+  condition: boolean,
+  product = '',
+  selected = '',
+): boolean {
+  if (product !== selected) {
+    return true;
+  }
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  if (condition) {
+    return false;
+  }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  return true;
+}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+    isSelected: true,
```

Comment:
why do you need this "isSelected" property? you can just use the selectedGood.
if this is not an empty string, thats mean that some Good is selected


this will simplify your code and the methods you created